automl/tests/unit/gapic/v1beta1/test_tables_client_v1beta1.py

```diff
@@ -0,0 +1,1365 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests."""
+
+import mock
+import pytest
+
+from google.cloud import automl_v1beta1
+from google.api_core import exceptions
+from google.cloud.automl_v1beta1.proto import data_types_pb2
+
+PROJECT = "project"
+REGION = "region"
+LOCATION_PATH = "projects/{}/locations/{}".format(PROJECT, REGION)
+
+
+class TestTablesClient(object):
+    def tables_client(self, client_attrs={}, prediction_client_attrs={}):
+        client_mock = mock.Mock(**client_attrs)
+        prediction_client_mock = mock.Mock(**prediction_client_attrs)
+        return automl_v1beta1.TablesClient(
+            client=client_mock,
+            prediction_client=prediction_client_mock,
+            project=PROJECT,
+            region=REGION,
+        )
+
+    def test_list_datasets_empty(self):
+        client = self.tables_client(
+            {
+                "list_datasets.return_value": [],
+                "location_path.return_value": LOCATION_PATH,
+            },
+            {},
+        )
+        ds = client.list_datasets()
+        client.auto_ml_client.location_path.assert_called_with(PROJECT, REGION)
+        client.auto_ml_client.list_datasets.assert_called_with(LOCATION_PATH)
+        assert ds == []
+
+    def test_list_datasets_not_empty(self):
+        datasets = ["some_dataset"]
+        client = self.tables_client(
+            {
+                "list_datasets.return_value": datasets,
+                "location_path.return_value": LOCATION_PATH,
+            },
+            {},
+        )
+        ds = client.list_datasets()
+        client.auto_ml_client.location_path.assert_called_with(PROJECT, REGION)
+        client.auto_ml_client.list_datasets.assert_called_with(LOCATION_PATH)
+        assert len(ds) == 1
+        assert ds[0] == "some_dataset"
+
+    def test_get_dataset_no_value(self):
+        dataset_actual = "dataset"
+        client = self.tables_client({}, {})
+        error = None
```

Comment:
Would it be possible to use `pytest.raises()` to check for expected exceptions? [pytest documentation](http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions)

```python
with pytest.raises(ValueError):
    client.get_dataset()
```

Reply:
Ah that looks much neater, I'll switch to that.